src/clj_puppetdb/http.clj

```diff
@@ -10,22 +11,38 @@
 ;;   - Validate schema for GET params. The GetParams schema
 ;;     exists, but needs work before it can be used.
 
-(defn make-https-client
-  [^String host {:keys [ssl-ca-cert ssl-cert ssl-key]}]
+(defn- make-https-client
+  [^String host {:keys [ssl-ca-cert ssl-cert ssl-key vcr-dir]}]
   {:pre  [(every? file? [ssl-ca-cert ssl-cert ssl-key])
           (.startsWith host "https://")]
    :post [(s/validate Client %)]}
   {:host host
    :opts {:ssl-ca-cert ssl-ca-cert
           :ssl-cert    ssl-cert
-          :ssl-key     ssl-key}})
+          :ssl-key     ssl-key
```

Comment:
While not directly related to this PR, it would better to pre-create an SSL context object instead of passing on the certificate/private key paths. (Something along these lines: https://github.com/mruzicka/pe-state-inspector-middleware/blob/ebf1e877ed281a945041b0ac3005b3063ea60bb4/src/clj/puppetlabs/state_inspector/middleware/core.clj#L26-L30.) The benefit of this approach is that the SSL context can be reused when establishing a new connection. On the contrary if the SSL context is not pre-created it has to be created anyways before a connection can be established but in this case it is not reused - it has to be created for each connection anew (see here https://github.com/mruzicka/pe-state-inspector-middleware/blob/ebf1e877ed281a945041b0ac3005b3063ea60bb4/src/clj/puppetlabs/state_inspector/middleware/core.clj#L26-L30).